src/mlpack/methods/ann/layer/multi_layer_impl.hpp

```diff
@@ -316,7 +316,6 @@ void MultiLayer<MatType>::ComputeOutputDimensions()
   inSize = this->inputDimensions[0];
   for (size_t i = 1; i < this->inputDimensions.size(); ++i)
     inSize *= this->inputDimensions[i];
-  totalInputSize += inSize;
```

Comment:
This is a bit tedious, but do you think we can rename `totalInputSize` to `totalDeltaSize`?  It no longer represents the full size of all the inputs to every layer, but instead just the total size required for the delta matrix for each layer.

Reply:
Note also that the comments in `multi_layer.hpp` for this and `totalOutputSize` are now incorrect and should also be changed (they aren't for every layer, they're for every layer except the first or last depending on which one we're talking about).